### PR TITLE
 Add DNS-SOA record.

### PR DESCRIPTION
**Pull request by Heidistein**
Bit different, this one returns a hash with the fields. This is a
 present from the Ruby resolver class.

### CODE REVIEW

File: lib/puppet/functions/dns_soa.rb [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+    param 'String', :record
+  end
+
+  dispatch :dns_soa_with_default do
+    param 'String', :record
+    block_param
+  end
+
+  def dns_soa(record)
+    Puppet.deprecation_warning('dns_soa', 'This method is deprecated please use the namspaced version dnsquery::soa')
```
Comment (smortex) [inline lib/puppet/functions/dns_soa.rb:19]: This method does not exist yet, so I think we can just remove this file :wink:

File: spec/functions/dns_soa_spec.rb [hunk 2]
```diff
@@ -0,0 +1,18 @@
+#! /usr/bin/env ruby -S rspec
```
Comment (smortex) [inline spec/functions/dns_soa_spec.rb:1]: This is not a valid shebang.  On Debian:

```
/usr/bin/env: 'ruby -S rspec': No such file or directory
/usr/bin/env: use -[v]S to pass options in shebang lines
```

Of course, the `-S` flag of `env(1)` is not portable: on FreeBSD:

```
env: ruby -S rspec: No such file or directory
```

So better remove the shebang (and make sure the file is not executable).

File: spec/functions/dnsquery_soa_spec.rb [hunk 3]
```diff
@@ -0,0 +1,18 @@
+#! /usr/bin/env ruby -S rspec
```
Comment (smortex) [inline spec/functions/dnsquery_soa_spec.rb:1]: Same here

File: spec/functions/dns_soa_spec.rb (hunk 2 above)
Comment (smortex) [reply spec/functions/dns_soa_spec.rb:1]: Also, this file can be removed because there is no such "legacy" function :smile:

**Review by smortex — COMMENTED** [review]
Looks good, only minor fixes.

File: lib/puppet/functions/dns_soa.rb (hunk 1 above)
Comment (Heidistein) [reply lib/puppet/functions/dns_soa.rb:19]: Yes, I hoped you wouldn't notice, I hacked it in my local module, and used this method. If I fix one deprecation, I should do them all, which is work ;-)

**Comment by bastelfreak** [comment]
thanks for the work!

**Comment by b4ldr** [comment]
thanks :)

**Pull request merged**